### PR TITLE
Intro Code

### PR DESCRIPTION
**Pull request by volts-inventory**
-Integrated michelle's old intro code
-Added entries in UI for DOB and Gender
-Adds UI information into user manager

### CODE REVIEW

**Comment by 0queue** [comment]
Initial notes: style changes, remove unnecessary xml (layouts, random strings), restore color blind mode

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,27 +1,28 @@
...
 
     <application
         android:allowBackup="true"
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
-        android:theme="@style/AppTheme.ColorBlind">
-        <activity android:name=".MainActivity">
+        android:theme="@style/AppTheme">
```
Comment (0queue) [inline app/src/main/AndroidManifest.xml]: Should be AppTheme.ColorBlind

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
+                <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
 
         <service
             android:name=".model.PharmacistService"
             android:enabled="true"
-            android:exported="false">
-        </service>
+            android:exported="false"></service>
```
Comment (0queue) [inline app/src/main/AndroidManifest.xml:22]: Let's get rid of these empty tags while we're at it

File: app/src/main/java/edu/umd/cmsc436/mstestsuite/IntroActivity.java [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    private TextView[] dots;
+    private LinearLayout dotsLayout;
+    private UserManager mUserManager;
+
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        mUserManager = new UserManager(this.getApplicationContext());
+        if (mUserManager.getCurUserID() != null) {
+            Log.i("Login", "There was a user");
```
Comment (0queue) [inline app/src/main/java/edu/umd/cmsc436/mstestsuite/IntroActivity.java]: Get rid of unnecessary logs

File: app/src/main/java/edu/umd/cmsc436/mstestsuite/IntroActivity.java [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        layouts = new int[]{R.layout.activity_intro_1, R.layout.activity_intro_2, R.layout.activity_intro_3};
+        addBottomDots(0);
+        viewPagerAdapter = new ViewPagerAdapter();
+        viewPager.setAdapter(viewPagerAdapter);
+        viewPager.addOnPageChangeListener(viewListener);
+
+        next.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                int current = getItem(+1);
```
Comment (0queue) [inline app/src/main/java/edu/umd/cmsc436/mstestsuite/IntroActivity.java]: That call to getItem(+1) can just be getItem(1)

File: app/src/main/java/edu/umd/cmsc436/mstestsuite/IntroActivity.java [hunk 5]
```diff
@@ -0,0 +1,190 @@
...
+            dots[position].setTextColor(colorActive[position]);
+        }
+    }
+
+    private void saveUser(){
+        EditText name = (EditText) findViewById(R.id.patient_id_box);
+        EditText dob = (EditText) findViewById(R.id.dob);
+        Spinner handSpin = (Spinner) findViewById(R.id.handedness_spinner);
+        Spinner genderSpin = (Spinner) findViewById(R.id.gender_spinner);
+        Log.i("Log", "USER Added");
```
Comment (0queue) [inline app/src/main/java/edu/umd/cmsc436/mstestsuite/IntroActivity.java]: (Log again)

File: app/src/main/res/layout/activity_intro.xml [hunk 6]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (0queue) [inline app/src/main/res/layout/activity_intro.xml:1]: Whole file could use auto-format (Ctrl-Alt-L, or something similar on Mac)

File: app/src/main/res/layout/activity_intro_1.xml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (0queue) [inline app/src/main/res/layout/activity_intro_1.xml:1]: (Auto format)

File: app/src/main/res/layout/activity_intro_2.xml [hunk 8]
```diff
@@ -0,0 +1,92 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (0queue) [inline app/src/main/res/layout/activity_intro_2.xml:1]: (Auto format)

File: app/src/main/res/layout/activity_intro_3.xml [hunk 9]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (0queue) [inline app/src/main/res/layout/activity_intro_3.xml]: This file should be combined with activity_intro_1, so there are only two intro layouts

File: app/src/main/res/values/colors.xml [hunk 10]
```diff
@@ -2,7 +2,29 @@
 <resources>
     <color name="colorPrimary">#F0163A</color>
     <color name="colorPrimaryDark">#CC1231</color>
-    <color name="colorAccent">#FFD90f</color>
+    <color name="colorAccent">#FFD90F</color>
     <color name="grayBackground">#9e9e9e</color>
 
+    <color name="screen1">#E3F2FD</color>
```
Comment (0queue) [inline app/src/main/res/values/colors.xml]: All of these new colors should probably be changed, to a black/white or very light gray, with the dots being colorAccent436 and colorAccentLight436

File: app/src/main/res/values/dimens.xml [hunk 11]
```diff
@@ -3,4 +3,10 @@
     <dimen name="bottom_sheet_elevation">10dp</dimen>
     <dimen name="practice_icon_size">120dp</dimen>
     <dimen name="practice_test_elevation">4dp</dimen>
+
+    <dimen name="activity_horizontal_margin">16dp</dimen>
+    <dimen name="activity_vertical_margin">16dp</dimen>
+    <dimen name="fab_margin">16dp</dimen>
```
Comment (0queue) [inline app/src/main/res/values/dimens.xml]: Only fab_margin is being used, and not even for a fab, so these should be removed

File: app/src/main/res/values/strings.xml [hunk 12]
```diff
@@ -10,4 +10,38 @@
...
+    <string name="welcome">Welcome to Test Suite</string>
+    <string name="intro_title">Test Suite</string>
+    <string name="patient_id">Patient ID:</string>
+    <string name="enter_patient_id">Enter Patient ID</string>
+    <string name="dob">Date of Birth(mm/dd/yyyy):</string>
+    <string name="gender">Gender</string>
+    <string name="handedness">Handedness:</string>
+    <string name="test_recyclerview_header">Practice Mode</string>
+
+    <string name="tap_start_text">Tap to start</string>
```
Comment (0queue) [inline app/src/main/res/values/strings.xml]: Unnecessary strings

File: app/src/main/AndroidManifest.xml [hunk 13]
```diff
@@ -1,27 +1,28 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-          package="edu.umd.cmsc436.mstestsuite">
+    package="edu.umd.cmsc436.mstestsuite">
```
Comment (0queue) [inline app/src/main/AndroidManifest.xml:3]: Should also be auto formatted

**Review by 0queue — CHANGES_REQUESTED** [review]
Clean up of some leftovers from the other project, and then style changes.  Also, combine the last and first intro layouts, they don't need to be separate

File: app/src/main/res/values/dimens.xml (hunk 11 above)
Comment (volts-inventory) [reply app/src/main/res/values/dimens.xml]: We need fab_margin

File: app/src/main/res/values/strings.xml (hunk 12 above)
Comment (volts-inventory) [reply app/src/main/res/values/strings.xml]: we need handedness

File: app/src/main/java/edu/umd/cmsc436/mstestsuite/IntroActivity.java [hunk 14]
```diff
@@ -13,12 +13,16 @@
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
+import android.view.inputmethod.InputMethodManager;
```
Comment (0queue) [inline app/src/main/java/edu/umd/cmsc436/mstestsuite/IntroActivity.java]: Unused imports (Ctrl-Alt-O on the file to remove)

File: app/src/main/java/edu/umd/cmsc436/mstestsuite/IntroActivity.java [hunk 15]
```diff
@@ -35,9 +39,11 @@
     @Override
     protected void onCreate(final Bundle savedInstanceState) {
```
Comment (0queue) [inline app/src/main/java/edu/umd/cmsc436/mstestsuite/IntroActivity.java]: Just one last style change: Change the status bar color to white with the line `getWindow().setStatusBarColor(ContextCompat.getColor(this, android.R.color.white));` (It looks really slick)

File: app/src/main/res/layout/activity_intro.xml [hunk 16]
```diff
@@ -1,40 +1,53 @@
...
+        android:layout_width="0dp"
+        android:layout_height="wrap_content"
+        android:background="?attr/colorPrimary"
+        android:minHeight="?attr/actionBarSize"
+        android:theme="@style/ToolbarTheme"
+        app:layout_constraintHorizontal_bias="1.0"
+        app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toRightOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
```
Comment (0queue) [inline app/src/main/res/layout/activity_intro.xml]: Not sure why there's a Toolbar here, it can be deleted

**Review by 0queue — CHANGES_REQUESTED** [review]
Looks a lot better, thanks!  Just three tweaks and that'll be it!

**Review by 0queue — APPROVED** [review]

**Pull request merged**